kubespawner/objects.py

```diff
@@ -150,6 +155,28 @@ def make_pod(
     notebook_container.lifecycle = lifecycle_hooks
     notebook_container.resources = V1ResourceRequirements()
     
+    if service_account is None:
+        # Add a hack to ensure that no service accounts are mounted in spawned pods
+        # This makes sure that we don"t accidentally give access to the whole
+        # kubernetes API to the users in the spawned pods.
+        # Note: We don't simply use `automountServiceAccountToken` here since we wanna be compatible
+        # with older kubernetes versions too for now.
+        hack_volume = V1Volume()
+        hack_volume.name =  "no-api-access-please"
+        hack_volume.empty_dir = {}
+        hack_volumes = [hack_volume]
+
+        hack_volume_mount = V1VolumeMount()
```

Comment:
This way of locking down access is rather odd. Why would the user want to disable API access this way? Ideally, I'd expect that the pod takes the default service account, and the user uses RBAC to lock down that service account.

Reply:
+1. In some cases having access to specific namespaces is useful to trigger other Operators for example.

Reply:
Right, I agree. This patch lets you disable this hack and set `c.KubeSpawner.singleuser_service_account = "default"` to have the service account be mounted. Using `automountServiceAccountToken` would make kubespawner be incompatible with pre-1.6 kubernetes versions, and since:

1. This hack is functionally still equivalent
2. There is nothing else that forces us to be 1.6+

I'd rather stick with this for now. However, if / when either of those conditions are false, am happy to bump required version up to 1.6 and get rid of this hack.